city_scrapers/spiders/il_sex_offender_management.py

```diff
@@ -0,0 +1,249 @@
+import json
+import re
+from datetime import datetime, timedelta
+from io import BytesIO, StringIO
+
+import scrapy
+from city_scrapers_core.constants import BOARD, CANCELLED, PASSED
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from pdfminer.high_level import extract_text_to_fp
+from pdfminer.layout import LAParams
+
+
+class IlSexOffenderManagementSpider(CityScrapersSpider):
+    name = "il_sex_offender_management"
+    agency = "Illinois Sex Offender Management Board"
+    timezone = "America/Chicago"
+    custom_settings = {"ROBOTSTXT_OBEY": False}
+    start_urls = [
+        "https://www2.illinois.gov/idoc/" + "Pages/SexOffenderManagementBoard.aspx"
+    ]
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+
+        for link in response.css(
+            "#ctl00_PlaceHolderMain_ctl01" + "__ControlWrapper_RichHtmlField a"
+        ):
+            link_text = " ".join(link.css("*::text").extract())
+            if "Meeting" not in link_text:
+                continue
+            if "Agenda" in link_text:
+                # scrape agenda pdf for meeting information
+                yield scrapy.Request(
+                    response.urljoin(link.attrib["href"]),
+                    callback=self._parse_documents,
+                    dont_filter=True,
+                )
+            # special case where link to video was uploaded rather than agenda
+            if "WebEx" in link_text:
+                yield scrapy.Request(
+                    link.attrib["href"], callback=self._parse_meta_video
+                )
+
+    def _parse_meta_video(self, response):
+        """Parse through meta of response in JSON if link provided rather than PDF"""
+        meta_text = response.css(
+            "[type='application/json']:not(#bootstrapData)::text"
+        ).extract_first()
+        if not meta_text:
+            return
+        meta_text = meta_text.strip()
+        clean_text_vid = json.loads(meta_text)
+
+        start_time, end_time = self._video_start_end_time(clean_text_vid)
+
+        meeting = Meeting(
+            title="SEX OFFENDER MANAGEMENT BOARD",
+            description="",
+            classification=BOARD,
+            start=start_time,
+            end=end_time,
+            all_day=False,
+            time_notes="",
+            location={
+                "address": response.urljoin(""),
+                "name": "WebEx Meeting (See meeting links to confirm)",
+            },
+            links={"title": "WebEx Meeting Link", "href": response.urljoin("")},
```

Comment:
`links` should always return a list (more info [in the docs](https://cityscrapers.org/docs/development/#links)), but in this case we can include the meeting link in `source` instead of `links` because it's where we parsed the content and not a separate attachment

Reply:
Would the meeting link be in addition to the original source (the website itself) or would it replace it?